engine/src/main/java/org/camunda/bpm/engine/impl/persistence/entity/JobEntity.java

```diff
@@ -142,7 +142,8 @@ else if (caseExecutionId != null) {
     JobHandler jobHandler = getJobHandler();
     JobHandlerConfiguration configuration = getJobHandlerConfiguration();
     ensureNotNull("Cannot find job handler '" + jobHandlerType + "' from job '" + this + "'", "jobHandler", jobHandler);
-    jobHandler.execute(configuration, execution, commandContext, tenantId);
+    if(execution!=null) jobHandler.execute(configuration, execution, commandContext, tenantId);
```

Comment:
In line 125 to 133 handles already the cases where the `execution` or `caseExecution` is set. To invoke the job handler the `context` is set. So that you only have to pass the `context` to the job handler like this:

```
jobHandler.execute(configuration, context, commandContext, tenantId);
```